src/Toolkit/Toolkit.WPF/Internal/MediaInlineControl.cs

```diff
@@ -0,0 +1,193 @@
+
+            var mainGrid = new Grid();
+            mainGrid.RowDefinitions.Add(new RowDefinition { Height = new GridLength(1, GridUnitType.Star) });
+            mainGrid.RowDefinitions.Add(new RowDefinition { Height = GridLength.Auto });
+
+            mainGrid.Children.Add(_mediaElement);
+            Grid.SetRow(_mediaElement, 0);
+
+            mainGrid.Children.Add(controlsGrid);
+            Grid.SetRow(controlsGrid, 1);
+
+            Content = mainGrid;
+
+            // Timer for updating slider and timestamp
+            _timer = new DispatcherTimer
+            {
+                Interval = TimeSpan.FromMilliseconds(200)
+            };
+            _timer.Tick += (s, e) =>
+            {
+                if (!_isSliderDragging && _duration.TotalSeconds > 0)
+                {
+                    _progressSlider.Value = _mediaElement.Position.TotalSeconds;
+                    UpdateTimestamp();
+                }
+            };
+
+            _playPauseButton.Click += (s, e) =>
+            {
+                if (_isPlaying)
+                {
+                    _mediaElement.Pause();
+                    _isPlaying = false;
+                    _playPauseButton.Content = PlayButtonGlyph;
+                    _timer.Stop();
+                }
+                else
+                {
+                    _mediaElement.Play();
+                    _isPlaying = true;
+                    _playPauseButton.Content = PauseButtonGlyph;
+                    _timer.Start();
+                }
+            };
+
+            _progressSlider.PreviewMouseDown += (s, e) => _isSliderDragging = true;
+            _progressSlider.PreviewMouseUp += (s, e) =>
+            {
+                if (_duration.TotalSeconds > 0)
+                {
+                    _mediaElement.Position = TimeSpan.FromSeconds(_progressSlider.Value);
+                }
+                _isSliderDragging = false;
+            };
+
+            _progressSlider.ValueChanged += (s, e) =>
+            {
+                if (_isSliderDragging && _duration.TotalSeconds > 0)
+                {
+                    _timestampText.Text = $"{FormatTime(TimeSpan.FromSeconds(_progressSlider.Value))} / {FormatTime(_duration)}";
+                }
+            };
+
+            _mediaElement.MediaOpened += (s, e) =>
+            {
+                _duration = _mediaElement.NaturalDuration.HasTimeSpan ? _mediaElement.NaturalDuration.TimeSpan : TimeSpan.Zero;
+                _progressSlider.Maximum = _duration.TotalSeconds;
+                UpdateTimestamp();
+            };
+
+            _mediaElement.MediaEnded += (s, e) =>
+            {
+                _mediaElement.Stop();
+                _isPlaying = false;
+                _playPauseButton.Content = PlayButtonGlyph;
+                _timer.Stop();
+                _progressSlider.Value = 0;
+                UpdateTimestamp();
+            };
+            _mediaElement.MediaFailed += (s, e) =>
```

Comment:
It might be helpful to log the original media error (`e.ErrorException`) using Trace.WriteLine, like we do for e.g. layer load errors.